container/rkt/factory.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rkt
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/google/cadvisor/container"
+	"github.com/google/cadvisor/fs"
+	info "github.com/google/cadvisor/info/v1"
+)
+
+// Basepath to all container specific information that libcontainer stores.
+var rktPath = flag.String("rkt_path", "/var/lib/rkt", "Absolute path to the Rkt root directory")
```

Comment:
If there is a plan to switch to the service, then how will the existing
method be modified to get the path from the service? We have the exact same
issue with docker factory :(

On Wed, Feb 10, 2016 at 3:59 PM, Shaya Potter <notifications@github.com>
wrote:

> In container/rkt/factory.go
> <https://github.com/google/cadvisor/pull/1105#discussion_r52547311>:
>
> > +// See the License for the specific language governing permissions and
> > +// limitations under the License.
> > +
> > +package rkt
> > +
> > +import (
> > +	"flag"
> > +	"fmt"
> > +
> > +	"github.com/google/cadvisor/container"
> > +	"github.com/google/cadvisor/fs"
> > +	info "github.com/google/cadvisor/info/v1"
> > +)
> > +
> > +// Basepath to all container specific information that libcontainer stores.
> > +var rktPath = flag.String("rkt_path", "/var/lib/rkt", "Absolute path to the Rkt root directory")
>
> I'm hoping to get this info from the rkt api-service at some point in
> time, but right now that service is limited.
> … <#-1722709039_>
> On Wed, Feb 10, 2016 at 3:53 PM, Vish Kannan <notifications@github.com>
> wrote: In container/rkt/factory.go <#1105 (comment)
> <https://github.com/google/cadvisor/pull/1105#discussion_r52546728>>: >
> +// See the License for the specific language governing permissions and >
> +// limitations under the License. > + > +package rkt > + > +import ( > +
> "flag" > + "fmt" > + > + "github.com/google/cadvisor/container" > + "
> github.com/google/cadvisor/fs" > + info "
> github.com/google/cadvisor/info/v1" > +) > + > +// Basepath to all
> container specific information that libcontainer stores. > +var rktPath =
> flag.String("rkt_path", "/var/lib/rkt", "Absolute path to the Rkt root
> directory") nit: The current code base is complex because flags are
> configured alongside regular code. Since you are adding new (and better)
> code, can we move the flags to a top level file and inject it into this
> package? So instead of providing a static method in this package, the path
> will only be returned from a rkt factory object. I intend to perform a
> similar surgery on the docker factory. Also, is it possible to auto-detect
> this path? — Reply to this email directly or view it on GitHub <
> https://github.com/google/cadvisor/pull/1105/files#r52546728>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1105/files#r52547311>.
>